plugins/parodos/src/components/workflow/hooks/workflowsPayload.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  WorkType,
+  type WorkflowDefinition,
+} from '../../../models/workflowDefinitionSchema';
+import { type WorkflowsPayload } from '../../../models/worksPayloadSchema';
+import get from 'lodash.get';
+import { type StrictRJSFSchema } from '@rjsf/utils';
+
+interface GetWorklfowsPayload {
+  workflow: WorkflowDefinition;
+  projectId: string;
+  schema: StrictRJSFSchema;
+}
+
+type Works = WorkflowsPayload['works'];
+type Work = WorkflowsPayload['works'][number];
+
+export function walkWorks(
+  works: WorkType[],
+  // TODO: improve type
+  formData: Record<string, any>,
+  prefix: string = '',
+): Works {
+  const result: Works = [];
+
+  for (const [index, work] of works.entries()) {
```

Comment:
nit: Array.map() ?

Reply:
@mareklibra I can't use map unfortunately because `walkWorks` is called recursively and weird stuff would happen if using map